runtime/doc/api.txt

```diff
@@ -2566,6 +2566,26 @@ nvim_win_get_var({window}, {name})                        *nvim_win_get_var()*
                 Return: ~
                     Variable value
 
+nvim_win_get_viewport({window})                      *nvim_win_get_viewport()*
+                Gets the current viewport of a window
+
+                The returned tuple contains (everything is 0-based,
+                end-exclusive):
+                • Top-most buffer line index
+                • Bottom-most buffer line index
+                • Left-most column index
+                • Right-most column index
```

Comment:
Just as a thought, but instead of `[top, bot, left, right]`, we instead have `[row, col, height, width]`, where:

* top  = row
* bot = row + height
* left = col
* right = col + width

Just so getting the dimensions of a viewport is more convenient as I often find myself wanting these more often than co-ordinates, otherwise the user can just do `height = top - bot` and `width = left - right`.

Not a big deal either way.

Reply:
Both are fine to me, and I trust your opinion as a plugin developer more seriously than mine as a little tree-sitter dev :wink: